### PR TITLE
fix linting, drop imports

### PR DESCRIPTION
**Pull request by Ichimonji10**
This pull request contains two commits because, although the two commits are logically separate, they both operate on module `robottelo.entities`. I could also submit two separate PRs, but that might slow down progress.

See the individual commit messages for full details on what's changed. Summary of changes:

* Fix linting issues in `robottelo.entities`. Pylint, sphinx and flake8 all parse the module without issues.
* Drop all robottelo imports from `robottelo.entities`. This makes it possible to extract the module from Robottelo.

Sample test results:

```
$ nosetests tests/foreman/api/test_multiple_paths.py
...S...S.....S................................................E............S...S.................F.....................................................EE...........................................................SSSS.SSSSSS..S.SSSSSSS...SSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
…
Ran 267 tests in 428.988s

FAILED (SKIP=53, errors=3, failures=1)
```

All failed and errored tests are due to known issues. `Host.create_missing` often provides incorrect values, and there's a versioning issue when calling `Product.read`.

### CODE REVIEW

**Comment by Ichimonji10** [comment]
This PR addresses #2065 and #2066.

**Comment by Ichimonji10** [comment]
I just had a great brainstorm on how to drop the last to robottelo imports in module `robottelo.entities`.

**Comment by Ichimonji10** [comment]
Rebased, rebased, and rebased again. :)

**Comment by elyezer** [comment]
The plan is to have all imports related to entities using the form `from robottelo import entities` and the next step will be changing it to `from nailgun import entities`, right?

ACK, pending clarification.

**Comment by Ichimonji10** [comment]
Yes! Here's the plan:

1. Copy `robottelo.entities` and `tests.robottelo.test_entities` to NailGun. Make minor changes to the files so that they work properly in their new home. Release version 0.8.0.
2. Update Robottelo by making all imports that reference `robottelo.entities` import `nailgun.entities` instead.

**Comment by elyezer** [comment]
ACK

**Comment by Ichimonji10** [comment]
For additional context, see https://github.com/SatelliteQE/robottelo/issues/979#issuecomment-93555748.

**Comment by Ichimonji10** [comment]
Bump.

Can I get this reviewed? I'd like to get this merged quickly so that I can file another PR.

File: robottelo/entities.py [hunk 1]
```diff
@@ -316,8 +337,7 @@ def create_missing(self):
         # generate an URL pointing to a docker server
         if provider.lower() == 'docker':
             if 'url' not in vars(self):
-                self.url = random.choice((
-                    get_internal_docker_url(), get_external_docker_url()))
+                self.url = self._server_config.url
```
Comment (omaciel) [inline robottelo/entities.py]: The ``url`` being generated here is not the same as what we expect for a docker-type repository (which is what the old code did.

Comment (elyezer) [inline robottelo/entities.py]: The expected format is `http[s]://server.com:port`

Comment (Ichimonji10) [inline robottelo/entities.py]: Shoot, you're right.

NailGun has no way of knowing about the settings in Robottelo's config file, such as the location of a docker compute resource. As a result, I've opted to make NailGun simply assume that a docker compute resource is available on the server. It's an assumption, but it's the best assumption available given the constraints on what knowledge NailGun has. I'll fix this to assign an address such as the following:

    self.url = '{0}:2325'.format(self._server_config.url)

**Comment by omaciel** [comment]
ACK pending comment about generating url for Docker-based repos.

File: robottelo/entities.py (hunk 1 above)
Comment (omaciel) [inline robottelo/entities.py]: :+1: for this solution.

**Comment by omaciel** [comment]
ACK

**Pull request merged**